src/runtime/pyscope.cs

```diff
@@ -0,0 +1,427 @@
+﻿using System;
+using System.Dynamic;
+
+namespace Python.Runtime
+{
+    public class PyScopeException : Exception
+    {
+        public PyScopeException(string message)
+            : base(message)
+        {
+
+        }
+    }
+
+    /// <summary>
+    /// Classes implement this interface must be used with GIL obtained.
+    /// </summary>
+    public interface IPyObject : IDisposable
+    {
+    }
+
+    public class PyScope : DynamicObject, IPyObject
+    {
+        public string Name
+        {
+            get;
+            private set;
+        }
+
+        private bool isDisposed;
+
+        internal PyScope(string name)
+        {
+            this.Name = name;
+            variables = Runtime.PyDict_New();
+            if (variables == IntPtr.Zero)
+            {
+                throw new PythonException();
+            }
+            Runtime.PyDict_SetItemString(
+                variables, "__builtins__",
+                Runtime.PyEval_GetBuiltins()
+            );
+            InitVariables();
+        }
+
+        //To compitable with the python module.
+        //Enable locals() and glocals() usage in the python script.
+        private void InitVariables()
+        {
+            SetVariable("locals", (Func<PyDict>)Variables);
+            SetVariable("globals", (Func<PyDict>)Variables);
+        }
+        
+        /// <summary>
+        /// the dict for scope variables
+        /// </summary>
+        internal IntPtr variables { get; private set; }
+
+        public long Refcount
```

Comment:
It's used for debug, now there are 2 IntPtr in the PyScope and the code seems stable, so it was removed.